### ISSUE TITLE
CollectionView in Toolkit Popup not work properly

### ISSUE THREAD

**Issue by FrancescoGiancristofaro** [dotnet/maui#20012]
### Description

CollectionView in a Popup not work correctly in these two situation:

- if property Size of Popup is settet
- if nested into a Frame

### Steps to Reproduce

Run attached project and try both Frame popup and Sized popup buttons

### Link to public reproduction project repository

https://github.com/FrancescoGiancristofaro/CollectionViewPopupIssue/tree/main

### Version with bug

8.0.6

### Is this a regression from previous behavior?

Yes, this used to work in Xamarin.Forms

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 14

### Did you find any workaround?

In sized popup case by removing size prop.
In frame popup case by removing frame and use an another control.

**Comment by ajay-mundi** [comment]
Honestly I just avoid using frames like the plague, I stick with border and things (generally) work well.

**Comment by XamlTest** [comment]
Verified this on VS 17.10.0 Preview 1.0(8.0.6). Repro on Android 14.0-API34, not repro on iOS 17.2 with below Project:
[CollectionViewPopupIssue.zip](https://github.com/dotnet/maui/files/14492120/CollectionViewPopupIssue.zip)

Android:
![image](https://github.com/dotnet/maui/assets/97647558/3700fd5a-9a2c-40f8-bf84-48fe06801b51)

**Comment by jsuarezruiz** [comment]
@jfversluis Should we move this one to the Toolkit repo?

**Comment by jfversluis** [comment]
Yeah seems like this is more a thing in Popup then CollectionView, lets move it there until we prove otherwise.

**Comment by jfversluis** [comment]
@FrancescoGiancristofaro I'm very sorry, but due to technical reasons (see linked above) we're not able to move this to the https://github.com/CommunityToolkit/Maui repo automatically. If this is still something that you're experiencing, could you maybe open an issue there as its more relevant to the Toolkit?

Thanks!
